### DISCUSSION TITLE
ncplane_boundlist clarification

### DISCUSSION THREAD

**Discussion by joseluis** [dankamongmen/notcurses#2202] [Q&A] [answered]
I was taking care of the bindings for `ncplane_boundlist` but I don't really understand its purpose, nor how it's supposed to be used. The doc comment just says: *"Gets the head of the list of planes bound to 'n'"*, and there's not much info anywhere else.

I see it exposes  `ncplane->blist;`. In the ncplane structure it says:
```
  // every plane is bound to some other plane, unless it is a root plane of a
  // pile. a pile has a set of one or more root planes, all of them siblings.
  // root planes are bound to themselves. the standard plane is always a root
  // plane (since it cannot be reparented). a path exists to a root plane.
```

My main questions are what is the "head of the list" exactly? Specially in relation with the root plane(s)? And what does it allow to do to have access to it?

**Comment by dankamongmen** [comment]
good eyes; this one was added at the last minute with some trepidation. it deserves some thinking.

**Comment by dankamongmen** [comment]
i'm almost certainly going to remove this ill-considered function.

**Comment by dankamongmen** [comment]
ok, yeah, this is gonna go. i'm gonna replace it with `ncplane_move_family_top()` and `ncplane_move_family_bottom()`, which will move a plane and its children to the specified location, retaining order among them. i.e. if you have:

A
B
C (binding E)
D
E

and call `ncplane_move_family_top()` on C, you'll get

C
E
A
B
D

`ncplane_move_family_bottom()` on C:

A
B
D
C
E

`ncplane_move_family_top()` on E:

E
A
B
C
D

`ncpane_move_family_bottom()` on E is a no-op.

**Comment by dankamongmen** [comment, marked as answer]
#2232 handles removing this accursed function.
